sympy/polys/rootoftools.py

```diff
@@ -970,6 +970,10 @@ def _eval_Eq(self, other):
         # is_real value of the CRootOf instance.
         if type(self) == type(other):
             return sympify(self == other)
+        if other.is_number and other.has(AppliedUndef) or not other.is_number:
```

Comment:
`other` will never have AppliedUndef if `other.is_number` is True:
```python
>>> f(1).is_number
False
```
So I think this should be
```python
if not other.is_number:
    return  # simplification by user might make it a number
```